rclcpp/include/rclcpp/executor.hpp

```diff
@@ -175,6 +175,23 @@ class Executor
   virtual void
   spin_some(std::chrono::nanoseconds max_duration = std::chrono::nanoseconds(0));
 
+  /// Collect and execute work repeatedly within a duration or until no more work is available.
+  /**
+   * This function can be overridden. The default implementation is suitable for a
+   * single-threaded model of execution.
+   * Adding subscriptions, timers, services, etc. with blocking callbacks will cause this function
+   * to block (which may have unintended consequences).
+   * If the time that waitables take to be executed is longer than the period on which new waitables
+   * become ready, this method will execute work repeatedly until `max_duration` has elapsed.
+   *
+   * \param[in] max_duration The maximum amount of time to spend executing work. Must be positive.
+   * Note that spin_some() may take longer than this time as it only returns once max_duration has
```

Comment:
Nit:

```suggestion
   * Note that spin_all() may take longer than this time as it only returns once max_duration has
```

Reply:
Ups, thanks for catching that. Fixed in https://github.com/ros2/rclcpp/pull/1156/commits/668a3b94fb769282a203fcfa85f428eb19dd6a4b.